ddl/db_test.go

```diff
@@ -5767,3 +5768,103 @@ func (s *testSerialDBSuite) TestColumnTypeChangeGenUniqueChangingName(c *C) {
 
 	tk.MustExec("drop table if exists t")
 }
+
+func (s *testSerialDBSuite) TestModifyColumnTypeWithWarnings(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	// Enable column change variable.
+	tk.Se.GetSessionVars().EnableChangeColumnType = true
+
+	// Test normal warnings.
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a decimal(5,2))")
+	tk.MustExec("insert into t values(111.22),(111.22),(111.22),(111.22),(333.4)")
+	// 111.22 will be truncated the fraction .22 as .2 with truncated warning for each row.
+	tk.MustExec("alter table t modify column a decimal(4,1)")
+	// there should 4 rows of warnings corresponding to the origin rows.
+	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1292 Truncated incorrect DECIMAL value: '111.22'",
```

Comment:
yes, there 2 points to notice there.
1: the row count is not precious in the parallel reorganization.
2: Not all the specific datum value can be stored in the warning for each row, only the first row of one warnings type with specific datum value does.

that means
```
Warning 1292 Truncated incorrect DECIMAL value: '111.22'
Warning 1292 Truncated incorrect DECIMAL value: '122.11' (won't be shown as  `value: '122.11'`, it will be seen as a copy the first one, because they are the same type)
```
So maybe the MySQL 5.7's warnings are more suitable for TiDB